### ISSUE TITLE
Remove pytest tests from clients package

### ISSUE THREAD

**Issue by kratsg** [rucio/rucio#4478]
```
$ python3 -m pip install rucio-clients -qqq
$ rucio whoami
Traceback (most recent call last):
  File "/usr/local/bin/rucio", line 75, in <module>
    from rucio.tests.test_rucio_server import TestRucioClient
  File "/usr/local/lib/python3.6/site-packages/rucio/tests/test_rucio_server.py", line 28, in <module>
    import pytest
ModuleNotFoundError: No module named 'pytest'
```

It seems to be that the latest release is forcing a call to `test_rucio_server` which is causing this error to occur. Can this be triaged quickly and probably get the latest release yanked from pypi?

- https://pypi.org/project/rucio-clients/1.25.1/

(Internal ATLAS reference to failed pipeline: https://gitlab.cern.ch/atlas-phys-susy-wg/Combinations/supersamples/-/jobs/12819064 )

**Comment by bari12** [comment]
Thanks Giordon for reporting this. 
The issue is due to us switching to `pytest` for testing (And the test files are included in the client package) this issue comes up if you do not have `pytest` in your environment.
We have to see how we fix this, probably we will remove the test files from the client package entirely.

For ATLAS workflows though it is recommended to use the rucio versions used in the ATLAS Local Root Base, since these are confirmed to work with ATLAS workflows and are tested more intensively in this context. 

I have yanked the release in the meanwhile.

**Comment by kratsg** [comment]
> The issue is due to us switching to `pytest` for testing (And the test files are included in the client package) this issue comes up if you do not have `pytest` in your environment.

yes, of course, but I would expect that `rucio` would declare `pytest` as a dependency if so (but I don't usually see `pytest` as a required user dependency, only more a `developer` dependency and additionally not shipping any testing in the python package.

**Comment by bziemons** [comment]
The issue was not introduced when switching to pytest, but here: https://github.com/rucio/rucio/blob/45afd73a0eb1be644194bb381deecc29923c6c65/lib/rucio/tests/test_rucio_server.py#L73 in fcf1f052c9af234b075b0f58be4c75b607588589

It was just overlooked by me that this file is used in the clients. I have prepared a fix by moving the test code in #4483
